AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -324,17 +337,31 @@ am_Error_e CAmSocketHandler::listenToSignals(const std::vector<uint8_t> & listSi
           /* We have a valid signal, read the info from the fd */
           struct signalfd_siginfo info;
           ssize_t bytes = read(pollfd.fd, &info, sizeof(info));
-          assert(bytes == sizeof(info));
-
+          if(bytes != sizeof(info))
+          {
+            //error received...
+            logError("Failed to read from signal fd");
+            throw std::runtime_error(std::string("Failed to read from signal fd."));
```

Comment:
Question: The previous version was throwing and exception only in DEBUG.
This is also working in case AM is build in RELEASE mode. Is it expected?